### ISSUE TITLE
Tests

### ISSUE THREAD

**Issue by arberiii** [libp2p/go-libp2p-nat#25]
Are tests needed for this package?

**Comment by Stebalien** [comment]
No. We should probably have unit tests but much of this can't be tested without actual NAT devices (or at least mock NAT devices).

Contributions of test cases would be appreciated.
